### DISCUSSION TITLE
Why doesn't the paper demonstrate more aggressive stunts, such as backflips?

### DISCUSSION THREAD

**Discussion by jgw365** [TinyMPC/TinyMPC#48] [Q&A] [answered]
Is it possible to achieve such maneuvers using TinyMPC? I was wondering why the paper didn't showcase more challenging actions like backflips, and whether TinyMPC is capable of handling these.

**Comment by plancherb1** [comment, marked as answer]
We focused on showing that we could support constraints in real-time and thus built to the dynamic obstacle avoidance demo. We did do the 90 degree off-axis recovering so that's 1/4 of the way there! :) There is no reason that you couldn't make a backflip work as well (and many folks have executed backflips on quadrotors in the past). I will note though that as per [https://github.com/orgs/TinyMPC/discussions/47](https://github.com/orgs/TinyMPC/discussions/47) we were using a single dynamics linearization which is strongly violated during a backflip and as such you would likely want to use at least two linearizations (right-side-up and upside-down) to better capture the dynamics during the maneuver.
